cockroachcloud/quickstart.md

```diff
@@ -327,76 +240,53 @@ You will see a welcome message when you've successfully connected to your cluste
     $ go run main.go
     ~~~
 
-    The program will prompt you for a connection string to the database:
+    The output will look like this:
 
     ~~~
-    Enter a connection string:
+    Hello world!
     ~~~
 
-1. Back in the **Connection info** dialog, click **Connection string**, copy the connection string from step 2, and paste it in your terminal after the "Enter a connection string" prompt.
+</section>
+<br/>{% comment %}Need to add this manual break to force Jekyll to render the next section correctly {% endcomment %}
 
-    {{site.data.alerts.callout_info}}
-    If the connection string does not include your SQL user password, replace `<ENTER-PASSWORD>` with the password.
-    {{site.data.alerts.end}}
+<section class="filter-content" markdown="1" data-scope="java">
 
-    The program will then execute. The output should look like this:
+1. Clone the `quickstart-code-samples` repo, and navigate to the `java` directory of the repo:
 
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    git clone https://github.com/cockroachdb/quickstart-code-samples
     ~~~
-    Hello world!
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    cd quickstart-code-samples/java
     ~~~
 
-</section>
-<br/>{% comment %}Need to add this manual break to force Jekyll to render the next section correctly {% endcomment %}
-<section class="filter-content" markdown="1" data-scope="java">
+    The code sample in this directory connects to {{ site.data.products.db }} with the [JDBC driver](https://jdbc.postgresql.org).
 
-1. Clone the following GitHub repository and check out the `serverless` branch:
+1. Use the `cockroach convert-url` command to convert the connection string that you [copied from the {{ site.data.products.cloud }} Console earlier](#step-2-connect-to-the-cluster) to a [valid connection string for JDBC connections](../stable/connect-to-the-database.html?filters=java):
 
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    git clone https://github.com/cockroachlabs/hello-world-java-jdbc.git
-    cd hello-world-java-jdbc
-    git checkout serverless
+    cockroach convert-url --url "<connection-string>"
```

Comment:
Kind of a downer that users need to download a 100+ MiB binary to convert a URL. That's software in 2022 for you!

Reply:
Yeaaaaaahhh... The JDBC connection string format is kind of a pain to deal with.